geo/src/algorithm/intersects/line_string.rs

```diff
@@ -3,31 +3,141 @@ use crate::BoundingRect;
 use crate::*;
 
 // Blanket implementation using self.lines().any().
-impl<T, G> Intersects<G> for LineString<T>
+macro_rules! blanket_intersects_linestring {
+    ($t:ty) => {
+        impl<T> $crate::Intersects<$t> for LineString<T>
+        where
+            T: GeoNum,
+            Line<T>: Intersects<$t>,
```

Comment:
We don't really need these constraints now that this isn't a real blanket implementation, right?